pkg/kudoctl/cmd/repo_remove.go

```diff
@@ -0,0 +1,68 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/kudobuilder/kudo/pkg/kudoctl/kudohome"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/repo"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type repoRemoveCmd struct {
+	out  io.Writer
+	name string
+	home kudohome.Home
+	fs   afero.Fs
+}
+
+func newRepoRemoveCmd(fs afero.Fs, out io.Writer) *cobra.Command {
+	remove := &repoRemoveCmd{out: out}
+
+	cmd := &cobra.Command{
+		Use:     "remove [flags] [NAME]",
+		Aliases: []string{"rm"},
+		Short:   "Remove an operator repository",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return fmt.Errorf("need at least one argument, name of operator repository")
+			}
+
+			remove.home = Settings.Home
+			remove.fs = fs
+			for i := 0; i < len(args); i++ {
+				remove.name = args[i]
+				if err := remove.run(); err != nil {
+					return err
+				}
+			}
+			return nil
+		},
+	}
+
+	return cmd
+}
+
+func (r *repoRemoveCmd) run() error {
+	return removeRepoLine(r.fs, r.out, r.name, r.home)
+}
+
+func removeRepoLine(fs afero.Fs, out io.Writer, name string, home kudohome.Home) error {
+	r, err := repo.LoadRepositories(fs, home.RepositoryFile())
+	if err != nil {
+		return err
+	}
+
+	if !r.Remove(name) {
+		return fmt.Errorf("no repo named %q found", name)
+	}
+	if err := r.WriteFile(fs, home.RepositoryFile(), 0644); err != nil {
```

Comment:
oh I wish all this handling of repository files was hidden in a package somewhere and we won't have to have permissions spread all over cmd and it would be nicely encapsulated somewhere